doc/source/serve/deployment-graph-cookbook.md

```diff
@@ -0,0 +1,72 @@
+---
+jupytext:
+  formats: ipynb,md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.6
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(serve-deployment-graph-cookbook)=
+
+# Deployment Graph Cookbook
+
+This doc is to provide more common use cases for using the deployment graph.
+
+# Catalog
+[Chain nodes with same class and different args](#chain_nodes_same_class_different_args)
+
+## Chain nodes with same class and different args <a name="chain_nodes_same_class_different_args"></a>
+
+The example shows how to chain nodes using the same class and having different args passed in
```

Comment:
I strongly suggest putting visualization here with node topology and intermediate output at each step, for all other snippets we put up. Something like 

(graph version)
   
    0               1                     2                     3
Input -> [Node_A] -> [Node_B] -> [Node_C] -> Output
                               0+1                1+2                 1+2+3